### ISSUE TITLE
Prop validator trigered on each update

### ISSUE THREAD

**Issue by miljan-aleksic** [vuejs/vue#8397]
### Version
2.5.16

### Reproduction link
https://jsfiddle.net/miljan/6hkf7znm/19/

### Steps to reproduce
Open the console log

### What is expected?
Console log should display 'validating foo' only once

### What is actually happening?
It is being logged (validated) each time any of the parent data is updated.

---
If a validator is heavy and trigered multiple times it can considerably slowdown the whole execution.

**Comment by Justineo** [comment]
It's expected behavior. Validators are designed to make sure you always have valid prop values. If you want to only validate props once, you can do this in the `created` hook.

**Comment by miljan-aleksic** [comment]
Yes, but what is the point of validating over and over the same value?

**Comment by Justineo** [comment]
Oh, sorry, it seems that I didn't quite understand the issue correctly earlier. An **unchanged** prop is being validated every time the parent is updated. I think we should further investigate on this to see if we can make some improvement.

**Comment by miljan-aleksic** [comment]
Exactly, I have updated the title with your resume :) Thanks.

**Comment by posva** [comment]
> If a validator is heavy and trigered multiple times it can considerably slowdown the whole execution

A validator shouldn't be that heavy 😄. Keep in mind they're only run in development mode to help developers provide correct data to components and make it clear that a prop is incorrect instead of seeing `Cannot read property foo of undefined` (and other non-readable js errors)
